src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -134,7 +134,7 @@ public void InstallWorkloads(IEnumerable<WorkloadId> workloadIds, bool skipManif
             if (!skipManifestUpdate)
             {
                 var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
-                if (File.Exists(installStateFilePath))
+                if (string.IsNullOrWhiteSpace(_fromRollbackDefinition) && File.Exists(installStateFilePath) && InstallStateContents.FromString(File.ReadAllText(installStateFilePath)).Manifests is not null)
```

Comment:
same comment as the other PR on useRollback.  I assume that any manifests is sufficient to know that the contents are pinned via rollback?  If one rollback usage pins 1 manifest and the next pins a different manifest, does the installstatefile end up with two entries or just the most recent?